setup.py

```diff
@@ -0,0 +1,11 @@
+import setuptools #enables develop
+
+setuptools.setup(
```

Comment:
I don't think this will work. Did you test this?

For pip packages the structure is
```
package_name/
    package_code/
    README.md
    LICENSE.md
    setup.py
```

So, for our case with two packages, we will have (within $EDGEML_HOME)

```
edgeml_tf/
    edgeml_tf/ # The package code
    LICENSE
    README.md
    setup.py
```    
and
```
edgeml_pytorch/
    edgeml_pytorch/ # The package code
    LICENSE
    README.md
    setup.py
```

Reply:
I overlooked this. I can change the name to setup_python since that needs to be released.

Do you want to release a TF package at this point?

Reply:
moved this to edgeml_pytorch

Reply:
Yeah we dont' have to release the tf package to `pypi`. Lets just stick to `pytorch`.

But that is not what I meant by structural change. Renaming `setup.py` doesnt have anything to do with that. 

The **confusion** here is because the parent directory has the same name as the package. Let me try to explain what I am trying to say with the old organization. Earlier we had the following structure:
```
${EDGEML_ROOT}/
|
|-- tf/                        # <--- this directory is where you run pip from
|-- edgeml_tf/                 # <--- This is the actual package. 
|-- setup.py                   # <--- This is the setup.py for that package.
|-- README.md        
```

Note how the `setup.py` and the top level directory of the package, that is the first part of `edgeml_tf.graph.*`, is on the same level.

In the new organization, the `tf` folder corresponds to the `edgeml_tf` folder and the `tf/edgeml_tf` folder corresponds to `edgeml_tf/edgeml_tf` folder. The code will be in `edgeml_tf/edgeml_tf/graph` and `edgeml_tf/edgeml_tf/trainer` etc.

Hope that makes sense. This is a pretty straighforward example for [reference](https://packaging.python.org/tutorials/packaging-projects/).

Reply:
@SachinG007  Could you have a look at this? Testing should fail in a new environment with this `setup.py`.

Reply:
@metastableB moving away from your hypothetical TF example to the scenario we actually want working, that is Pytorch, are you suggesting that we create another folder $EDGEML_ROOT/pytorch into which edgeml_pytorch goes?  One alternate is to leave the directory as is, and have setup in the root, and manually list the folders it is support to package.

Reply:
@metastableB  see last commit

Reply:
> are you suggesting that we create another folder $EDGEML_ROOT/pytorch into which edgeml_pytorch goes

Yes. For a few reasons.

1) Even though we are only **releasing** pytorch package and want to move forward with pytorch, I still want to keep the tensorflow package in the repository. TF will definitely become useful sometime in the future for RNN related work. One simple scenario that we will immediately need is testing/benchmarking tf and pytorch code within the same virtualenvironment. This will require local installations of both the packages. For this to work, we need two `setup.py` files. For that, both of them need to be in `$EDGEML_ROOT/pytorch/setup.py` and `$EDGEML_ROOT/tf/setup.py`.

2) As of now, our edgeml repository is a collection of various sub-projects; C++, tensorflow package, pytorch package, applications. In the future, other packages might pop up that would require its own `setup.py` files. Lets just be future proof and isolate `edgeml_pytorch` to its own directory. Lets not put any sub-project specific files at the top level directory ($EDGEML_ROOT). Else we will have to deal with this re-org again.


@adityakusupati  Can you have a look here? Do you have other suggestions?

Reply:
@metastableB, @harsha-simhadri  I don't understand python packaging and the norms that well.

1) I completely agree with @metastableB  on TF retaining its packaging structure until pytorch is viable for custom RNNs.
2) I again agree with you, given the possibility of more tools/directories/frameworks which might come up, it is better to isolate installation of each of the packages. 

However, having said the above two, I tried the old packaging format that is [here](https://github.com/pushkalkatara/EdgeML/tree/414273dd1181063197ab199ecc5d157fae06a0d2) and this works seamlessly for pytorch (I am not sure about TF as I haven't tested it) @SachinG007 and @pushkalkatara did the structure [here](https://github.com/pushkalkatara/EdgeML/tree/414273dd1181063197ab199ecc5d157fae06a0d2) ensure TF examples ran without issues?

Reply:
@adityakusupati That is odd. The pytoch link should not work, unless you run `pip install /.` from `${EDGEML_ROOT}`. If you try to install the package from within the `edgeml_pytorch` folder, pip will complain that `setup.py` is not found.

Anyway, @harsha-simhadri  the only thing you have to do is move `$EDGEML_ROOT/edgeml_pytorch` to `$EDGEML_ROOT/pytorch/edgeml_pytorch/` and similarly move the `$EDGEML_ROOT/edgeml_tf/` to `$EDGEML_ROOT/tf/edgeml_tf/`. Having done that, you then move `$EDGEML_ROOT/setup.py` to `$EDGEML_ROOT/pytorch/`. And create a setup file for the `tf` directory.

Reply:
OK, will create subfolders for TF and PyTorch. I wont create a TF set up file yet since I have not read through that code much.

Reply:
I'll raise a PR for that this weekend. Thanks.